$NetBSD: patch-src_glx_Makefile.am,v 1.3 2019/01/19 23:21:43 tnn Exp $

- We don't need libmesautil.la here. Those symbols seem to already be
  included in libmesa.la.
- force libtool to use clang++ to link libGL on OSX.
  Otherwise we get:
  Undefined symbols for architecture x86_64:
  "std::terminate()", referenced from:
      ___clang_call_terminate in libglx.a(glsl_parser.o)
      ...

--- src/glx/Makefile.am.orig	2019-01-17 11:26:23.000000000 +0000
+++ src/glx/Makefile.am
@@ -89,10 +89,16 @@ libglx_la_SOURCES = \
 	singlepix.c \
 	vertarr.c
 
+if HAVE_APPLEDRI
+libglx_la_LIBADD = \
+	$(top_builddir)/src/loader/libloader.la \
+	$(top_builddir)/src/util/libxmlconfig.la
+else
 libglx_la_LIBADD = \
 	$(top_builddir)/src/loader/libloader.la \
 	$(top_builddir)/src/util/libmesautil.la \
 	$(top_builddir)/src/util/libxmlconfig.la
+endif
 
 if HAVE_DRISW
 libglx_la_SOURCES += \
@@ -136,6 +142,7 @@ libglx_la_SOURCES += \
 
 SUBDIRS += apple
 libglx_la_LIBADD += $(builddir)/apple/libappleglx.la
+libglx_la_LIBADD += $(top_builddir)/src/mesa/libmesa.la
 endif
 
 if HAVE_WINDOWSDRI
@@ -187,6 +194,10 @@ lib@GL_LIB@_la_SOURCES =
 lib@GL_LIB@_la_LIBADD = $(GL_LIBS)
 lib@GL_LIB@_la_LDFLAGS = $(GL_LDFLAGS)
 
+if HAVE_APPLEDRI
+lib@GL_LIB@_la_SOURCES += apple_dummy.cpp
+endif
+
 if HAVE_WINDOWSDRI
 lib@GL_LIB@_la_LDFLAGS += -lgdi32 -lopengl32 -Wl,--disable-stdcall-fixup
 endif
